project-koku/koku
# [COST-4067] Include finalized-date on object metadata for HCS finalized files

## Jira Ticket

[COST-4067](https://issues.redhat.com/browse/COST-4067)

## Description

This change will add a finalized-date field to object metadata when HCS files are finalized. The finalization date is always the 15th of the month following the processed date, even on manual resummary. 

This PR also removes the `copy_hcs_data_to_s3_bucket` function in favor of passing metadata into `copy_data_to_s3_bucket` function. 

## Testing

1. Checkout Branch
2. Set `ENABLE_HCS_DEBUG` to True in your environment
3. Restart Koku
4. Load AWS data that correlates with HCS: `make create-test-customer` followed by `make load-test-customer-data test_source=AWS`
5. Check minio for an AWS source with data and look at an object in the bucket for last month. In the metadata, you should see `X-Amz-Metadata-Finalized` as False and nothing for `X-Amz-Metadata-Finalized-Date`. 
6. Run HCS finalization for last month manually via masu, replacing your provider_uuid here with the correct one:
```
http://localhost:5042/api/cost-management/v1/hcs_report_finalization/?provider_uuid=40b18d6d-5823-4e9c-8d50-d3d4eb1a43e3
```
7. Re-check the object in minio and you should see `X-Amz-Metadata-Finalized` is now True and `X-Amz-Metadata-Finalized-Date` is now set to 2023-08-15.

## Notes

...
